app/src/main/java/com/hover/stax/data/repository/AccountRepositoryImpl.kt

```diff
@@ -40,13 +40,13 @@ class AccountRepositoryImpl(val accountRepo: AccountRepo, val channelRepo: Chann
     override suspend fun createAccount(channel: Channel, simSubscriptionId: Int, isDefault: Boolean) {
         val accountName: String = if (getFetchAccountAction(channel.id) == null) channel.name else PLACEHOLDER //placeholder alias for easier identification later
         val account = Account(
-            accountName, channel.name, channel.logoUrl, channel.accountNo, channel.id, channel.institutionType, channel.countryAlpha2,
```

Comment:
Why change the id to the institution? That is certainly going to break something.

Also, why is this using the default constructor instead of a constructor that you pass the channel to?

Reply:
I looked at the Account's model the parameter expects institutionId, where it was previously passed channelId.

I can change back, but I suppose we need to attend it again.

Reply:
ah I see. But this points to why I will keep bringing up the constructor: there is already a constructor that takes channel as an argument and deals with the details correctly. Why aren't you using that one?